### PR TITLE
Handle charset as part of Content-Type headers

### PR DESCRIPTION
**Pull request by seanf**
https://zanata.atlassian.net/browse/ZNTA-1011

### CODE REVIEW

**Comment by carlosmunoz** [comment]
:+1:

File: zanata-rest-client/src/main/java/org/zanata/rest/client/InvalidContentTypeFilter.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
     // we assume only xml or json are the valid types (wildcard type is also considered compatible)
     private static final Pattern VALID_TYPES_REGEX =
-            Pattern.compile("application/.*\\+?(\\*|xml|json)");
+            Pattern.compile("application/.*\\+?(\\*|xml|json)(;charset=.*)?");
```
Comment (seanf) [inline zanata-rest-client/src/main/java/org/zanata/rest/client/InvalidContentTypeFilter.java]: Thanks @definite - I've made the regex less strict.

**Comment by definite** [comment]
:+1:

**Comment by seanf** [comment]
@carlosmunoz Happy with https://github.com/zanata/zanata-client/pull/110/commits/37e5d70627527f0a3b223dd2c3a3da2614c9f86a ?

**Comment by carlosmunoz** [comment]
@seanf I'm happy.
(jsut opened the regex to allow anything after the ;)

**Pull request merged**